src/toil/wdl/wdltoil.py

```diff
@@ -470,11 +485,21 @@ def _devirtualize_filename(self, filename: str) -> str:
         if filename.startswith(TOIL_URI_SCHEME):
             # This is a reference to the Toil filestore.
             # Deserialize the FileID
-            file_id, file_basename = unpack_toil_uri(filename)
+            file_id, parent_id, file_basename = unpack_toil_uri(filename)
 
             # Decide where it should be put
-            file_dir = self._file_store.getLocalTempDir()
-            dest_path = os.path.join(file_dir, file_basename)
+            if self._is_uuid(parent_id):
+                # This is a URI with the "parent" UUID attached to the filename
+                # Use UUID as folder name rather than a new temp folder to reduce internal clutter
```

Comment:
So the algorithm is to set up all the files from a particular previous job in a single host directory, under their basenames, and then mount them into the container so that everything in a given host directory stays together.

That definitely will mostly work, but it won't handle the case where multiple files with the same basename, but different paths, are uploaded by a job.

I can have a job make `dir1/index.txt` and `dir2/index.txt` as outputs, and use them both as inputs to another job. It looks like with the current implementation one of those is going to overwrite the other, since they have the same creator UUID and the same basename.